pandas/tests/groupby/test_groupby.py

```diff
@@ -1994,3 +1994,22 @@ def test_dup_labels_output_shape(groupby_func, idx):
 
     assert result.shape == (1, 2)
     tm.assert_index_equal(result.columns, idx)
+
+
+def test_groupby_crash_on_nunique(axis):
+    # Fix following 30253
+    df = pd.DataFrame(
+        {("A", "B"): [1, 2], ("A", "C"): [1, 3], ("D", "B"): [0, 0]},
+        index=pd.bdate_range("20191212", "20191213"),
+    )
+    df.columns.names = ["COL1", "COL2"]
+
+    axis_number = df._get_axis_number(axis)
+    if not axis_number:
+        df = df.T
+
+    nunique_t = df.T.groupby(level=0, axis=axis).nunique().T
+
+    nunique = df.groupby(axis=int(not axis_number), level=0).nunique()
+
+    pd.testing.assert_frame_equal(nunique_t, nunique)
```

Comment:
```suggestion
    tm.assert_frame_equal(result, expected)
```

Note will have to update variable names as well

Reply:
Sure, will do, thanks